src/shared/utils.ts

```diff
@@ -1,3 +1,82 @@
+import { NextRequest } from 'next/server'
 import './utils.css'
+import { AxiosResponse } from 'axios'
 // Had to be done this way because tailwind css is not dynamically rendering the image on build
 export const BG_IMG_CLASSES = 'background-logo'
+
+export interface BackendEventByTopic0ResponseValue {
+  address: string
+  blockNumber: string
+  data: string
+  gasPrice: string
+  gasUsed: string
+  logIndex: string
+  timeStamp: string
+  topics: Array<null | string>
+  transactionHash: string
+  transactionIndex: string
+}
+
+export type CachedData = {
+  lastUpdated: number
+  data: BackendEventByTopic0ResponseValue[]
+  isFetching: boolean
+  error: string
+  lastFromBlock: number
+}
+
+export const handleCachedGetRequest = (
```

Comment:
We discussed it with Antonio, since the issue is that the service is intermittent. It returns data and on some occasions it does not. 

We may discuss if you want, this is the same issue that we have in the proposals, this is why on some occasions there are no proposals returned, since the service keeps failing.